sdk/storage/Azure.Storage.Files/src/DirectoryClient.cs

```diff
@@ -41,6 +41,71 @@ public class DirectoryClient
         /// </summary>
         protected virtual HttpPipeline Pipeline => _pipeline;
 
+        /// <summary>
+        /// The Storage account name corresponding to the directory client.
+        /// </summary>
+        private string _accountName;
+
+        /// <summary>
+        /// Gets the Storage account name corresponding to the directory client.
+        /// </summary>
+        public string AccountName
+        {
+            get
+            {
+                SetNameFieldsIfNull();
+                return _accountName;
+            }
+        }
+
+        /// <summary>
+        /// The share name corresponding to the directory client.
+        /// </summary>
+        private string _shareName;
+
+        /// <summary>
+        /// Gets the share name corresponding to the directory client.
+        /// </summary>
+        public string ShareName
+        {
+            get
+            {
+                SetNameFieldsIfNull();
+                return _shareName;
+            }
+        }
+
+        /// <summary>
+        /// The name of the directory.
+        /// </summary>
+        private string _name;
+
+        /// <summary>
+        /// Gets the name of the directory.
+        /// </summary>
+        public string Name
```

Comment:
Should we include `string Path` in addition to the name?  For this and `FileClient`?  I'm kind of on the fence.

Reply:
What would the path contain? Everything except the last piece of DirectoryOrFilePath? Or just the entire path including the last piece?

Reply:
I was thinking the entire DirectoryOrPath segment.  Let's leave it off for now because customers can get it themselves and we can add a convenience in the future if we see the need.